### DISCUSSION TITLE
Does Zustand solve the stale props problem?

### DISCUSSION THREAD

**Discussion by jaskp** [pmndrs/zustand#2722] [Q&A]
Hi, one of the first sentences in the readme is that Zustand handles the zombie child problem, react concurrency, and context loss between mixed renderers. But does Zustand suffer from stale props like [`use-context-selector`](https://github.com/dai-shi/use-context-selector?tab=readme-ov-file#limitations) or is that handled as well?

Thanks!

**Comment by dai-shi** [comment]
I thought stale props and zombie children refer the same thing, no?

With automatic batching in React 18, we shouldn't suffer from stale props.

**Reply by jaskp** [reply]
@dai-shi

> I thought stale props and zombie children refer the same thing, no?

Well, I'm confused, because most of my knowledge in this area comes from things you wrote or linked to 😄 Specifically, you've linked to [this part of the Redux docs](https://react-redux.js.org/api/hooks#stale-props-and-zombie-children) that describes stale props and zombie children as two issues. Are they the same then?

---

> With automatic batching in React 18, we shouldn't suffer from stale props.

Then the `use-context-selector` docs are just outdated? I checked and this line was added back in [2019](https://github.com/dai-shi/use-context-selector/commit/4bcbdeff9e8e73a355ff903b047254369d8b0c97):

> The [stale props](https://react-redux.js.org/api/hooks#stale-props-and-zombie-children) issue can't be solved in userland.

**Reply by dai-shi** [reply]
I'd say, stale props can cause zombie children.

> Then the use-context-selector docs are just outdated?

Good catch actually.
https://github.com/dai-shi/use-context-selector/blob/ee95bff780b7069f856e654460b2448a04e0edb6/tests/03_stale_props.spec.tsx#L6
This test passes with React 18.
I'll update it.

**Reply by dai-shi** [reply]
https://github.com/dai-shi/use-context-selector/pull/137
